kv/dist_sender.go

```diff
@@ -555,32 +548,52 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 			var replicas replicaSlice
 			if err == nil {
 				replicas = newReplicaSlice(ds.gossip, desc)
-				rpcErr = ds.sendRPC(desc.RaftID, replicas, args, reply)
-				err = rpcErr
-			}
 
-			// If the RPC went through fine, an error (if any) is in the reply.
-			if err == nil {
-				err = reply.Header().GoError()
+				// Rearrange the replicas so that those replicas with long common
```

Comment:
Why not just have one `if lookupErr == nil` section where you do both leader lookup and optimizing order / sending the RPC.